app/validators/DateValidator.scala

```diff
@@ -30,43 +32,43 @@ trait DateValidator {
   val month = "month"
   val day = "day"
 
-  val rasDateConstraint : Constraint[RasDate] = Constraint("dateOfBirth") ({
+  def rasDateConstraint(name: String): Constraint[RasDate] = Constraint("dateOfBirth")({
     date => {
 
       val leapYear =
-        try{
+        try {
           new DateTime().withYear(date.year.getOrElse("0").toInt).year.isLeap
         } catch {
-          case e:NumberFormatException => false
+          case e: NumberFormatException => false
         }
 
       if (date.day.isEmpty && date.month.isEmpty && date.year.isEmpty)
-        Invalid(Seq(ValidationError(Messages("error.mandatory", Messages("dob")), day)))
+        Invalid(Seq(ValidationError(Messages("error.withName.mandatory", name, Messages("dob")), day)))
 
       else if (date.day.isEmpty)
-        Invalid(Seq(ValidationError(Messages("error.mandatory", Messages(day)), day)))
+        Invalid(Seq(ValidationError(Messages("error.withName.mandatory", name, Messages("dob")), day)))
```

Comment:
I'm going to try and get something confirmed for this. The changes here mean that the error message "Enter a day" (also applicable for month and year) will now get replaced with the same message. I don't know whether accessibility wise this is an issue, and in in general just clarity wise.